pypeds/example/model/regulation.py

```diff
@@ -38,8 +38,23 @@ def exert_single(self, source, target):
         force = dirt * (self.A * math.exp(-dist) / self.B)
         affection = Affection("Force", force)
         target.affected(affection)
-        if isinstance(target, Ellipse2D):
-            return
+        if isinstance(target.shape, Ellipse2D):
```

Comment:
shall not add this condition, instead torque will be calculated when a variable such as `cal_torque` is set to True. We do not care whether the shape is an ellipse or not. Just exert the torque on the agent and let it judge